From 84f32f4001fc6bdcc125ccc959081de022d18b6d Mon Sep 17 00:00:00 2001
From: Tony Clayton <tony.clayton@gmail.com>
Date: Fri, 8 Sep 2017 14:55:52 -0400
Subject: [PATCH] Fix for: libpam4j authenticates invalid accounts
 [https://github.com/kohsuke/libpam4j/issues/18]

Change-Id: I5052cd0d9f617239a63fb036daa697cbae5bb9d7
---
 src/main/java/org/jvnet/libpam/PAM.java | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/main/java/org/jvnet/libpam/PAM.java b/src/main/java/org/jvnet/libpam/PAM.java
index eb53ece..bd8d934 100644
--- a/src/main/java/org/jvnet/libpam/PAM.java
+++ b/src/main/java/org/jvnet/libpam/PAM.java
@@ -124,8 +124,7 @@ public UnixUser authenticate(String username, String password) throws PAMExcepti
             check(libpam.pam_set_item(pht,PAM_USER,username),"pam_set_item failed");
             check(libpam.pam_authenticate(pht,0),"pam_authenticate failed");
             check(libpam.pam_setcred(pht,0),"pam_setcred failed");
-            // several different error code seem to be used to represent authentication failures
-//            check(libpam.pam_acct_mgmt(pht,0),"pam_acct_mgmt failed");
+            check(libpam.pam_acct_mgmt(pht,0),"pam_acct_mgmt failed");
 
             PointerByReference r = new PointerByReference();
             check(libpam.pam_get_item(pht,PAM_USER,r),"pam_get_item failed");
